src/manifold/src/constructors.cpp

```diff
@@ -317,95 +317,127 @@ Manifold Manifold::Extrude(const CrossSection& crossSection, float height,
  * @param crossSection A set of non-overlapping polygons to revolve.
  * @param circularSegments Number of segments along its diameter. Default is
  * calculated by the static Defaults.
+ * @param revolveDegrees Number of degrees to revolve. Default is 360 degrees.
  */
 Manifold Manifold::Revolve(const CrossSection& crossSection,
-                           int circularSegments) {
-  auto polygons = crossSection.ToPolygons();
+                           int circularSegments,
+                           float revolveDegrees) {
+
+  Polygons polygons = crossSection.ToPolygons();
+
   if (polygons.size() == 0) {
     return Invalid();
   }
 
+  Rect bounds = crossSection.Bounds();
+
+  // Take the x>=0 slice.
+  if (bounds.min.x < 0) {
+    glm::vec2 min = bounds.min;
+    glm::vec2 max = bounds.max;
+    CrossSection posBoundingBox = CrossSection({{0.0, min.y},{max.x, min.y},
+                                                {max.x,max.y},{0.0,max.y}});
+
+    // Can't use RectClip as it has many failure cases.
+    polygons = (crossSection ^ posBoundingBox).ToPolygons();
+  }
+
   float radius = 0.0f;
   for (const auto& poly : polygons) {
     for (const auto& vert : poly) {
       radius = fmax(radius, vert.x);
     }
   }
+
+  if (revolveDegrees > 360.0f) {
+    revolveDegrees = 360.0f;
+  }
+  bool isFullRevolution = revolveDegrees == 360.0f;
+
   int nDivisions = circularSegments > 2 ? circularSegments
                                         : Quality::GetCircularSegments(radius);
+
   auto pImpl_ = std::make_shared<Impl>();
   auto& vertPos = pImpl_->vertPos_;
   VecDH<glm::ivec3> triVertsDH;
   auto& triVerts = triVertsDH;
-  float dPhi = 360.0f / nDivisions;
+
+  std::vector<int> startPoses;
+  std::vector<int> endPoses;
+
+  float dPhi = revolveDegrees / nDivisions;
```

Comment:
I've been trying to update our style to use `const` as much as possible; would you mind applying that across your PR?

Reply:
Yep, good call.